### PR TITLE
Palindrome Assignment Version 1

### PR DESCRIPTION
**Pull request by rochaalexandre**
- [x]  "mom" returns true
- [x] "Mom" returns true
- [x] "MoM" returns true
- [x] "Momx" returns false
- [x] "xMomx" returns true
- [x] "Was It A Rat I Saw" returns true
- [x] "Never Odd or Even" returns true
- [x] "Never Odd or Even1" returns false 
- [x] "1Never Odd or Even1" returns true

### CODE REVIEW

**Comment by taha-programmer** [comment]
Hi @rochaalexandre 

Well done on this assignment 👍🏼, I can see that you covered all the tests in the grading checklist.

a few things I'd do differently though:

I'd avoid mentioning technical words in my test name, like "return true", so I'd use 
```typescript
it('knows that "1Never Odd or Even1" is a palindrome',  () => {
        expect(isPalindrome('1Never Odd or Even1')).toBeTruthy()
    });
```
instead of
```typescript
it('should return true for "1Never Odd or Even1"', async () => {
        expect(isPalindrome('1Never Odd or Even1')).toBeTruthy()
    });
```
I'd also separate your tests based on the high-level behavior you're testing, so you could add a "describe" or it.each() block and inside it add tests for the high-level behavior of **Recognizing small-letter word palindromes** under which you indent it() tests for different success and failure examples. and add another "describe" block for testing the high-level behavior of "**Recognizing palindromes even when the casing is off**", and another one for the "**Recognizing palindromes of several words**", etc... under each high-level behavior include several success and failure examples. when you have many it() or test() targeting the same behavior you can refactor your test code and use it.each() to combine them together.

something like

```typescript
describe("palindrome checker", () => {
  describe("Recognizing simple palindrome words", () => {
// when you have many it() tests, use it.each()
    it("it knows that 'mom' is a palindrome", () => {
      expect(isPalindrome("mom")).toBe(true);
    });
    it("it knows that 'wow' is a palindrome", () => {
      expect(isPalindrome("wow")).toBe(true);
    });

    //Also include failure examples for this high-level behavior
    it("it knows that 'bill' is NOT a palindrome", () => {
      expect(isPalindrome("bill")).toBe(false);
    });
  });
});
```

File: 1_The_Physical/Getting_Started_Emergent_Design/1_Red_Green_Refactor/1_2_Palindrome/src/index.spec.ts [hunk 1]
```diff
@@ -1,4 +1,47 @@
...
+
+    it('should return true for "Never Odd or Even"', async () => {
+        expect(isPalindrome('Never Odd or Even')).toBeTruthy()
+    });
+
+    it('should return false for "Never Odd or Even1"', async () => {
+        expect(isPalindrome('Never Odd or Even1')).toBeFalsy()
+    });
+
+    it('should return true for "1Never Odd or Even1"', async () => {
```
Comment (taha-programmer) [inline 1_The_Physical/Getting_Started_Emergent_Design/1_Red_Green_Refactor/1_2_Palindrome/src/index.spec.ts:40]: I'm also curious to know why you used `async` in all of your test functions.
 
as far as I know, we only use async/await when dealing with asynchronous operations, such as fetching data from an API or reading a file from the disk.

 do you think we need it for this type of function here?

Comment (rochaalexandre) [reply 1_The_Physical/Getting_Started_Emergent_Design/1_Red_Green_Refactor/1_2_Palindrome/src/index.spec.ts:40]: This is only my IDE template, I haven't noticed until now that you pointed it out.

**Comment by rochaalexandre** [comment]
> Hi @rochaalexandre 
> 
> Well done on this assignment 👍🏼, I can see that you covered all the tests in the grading checklist.
> 
> a few things I'd do differently though:
> 
> I'd avoid mentioning technical words in my test name, like "return true", so I'd use 
> ```typescript
> it('knows that "1Never Odd or Even1" is a palindrome',  () => {
>         expect(isPalindrome('1Never Odd or Even1')).toBeTruthy()
>     });
> ```
> instead of
> ```typescript
> it('should return true for "1Never Odd or Even1"',  () => {
>         expect(isPalindrome('1Never Odd or Even1')).toBeTruthy()
>     });
> ```
> I'd also separate your tests based on the high-level behavior you're testing, so you could add a "describe" or it.each() block and inside it add tests for the high-level behavior of **Recognizing small-letter word palindromes** under which you indent it() tests for different success and failure examples. and add another "describe" block for testing the high-level behavior of "**Recognizing palindromes even when the casing is off**", and another one for the "**Recognizing palindromes of several words**", etc... under each high-level behavior include several success and failure examples. when you have many it() or test() targeting the same behavior you can refactor your test code and use it.each() to combine them together.
> 
> something like
> 
> ```typescript
> describe("palindrome checker", () => {
>   describe("Recognizing simple palindrome words", () => {
> // when you have many it() tests, use it.each()
>     it("it knows that 'mom' is a palindrome", () => {
>       expect(isPalindrome("mom")).toBe(true);
>     });
>     it("it knows that 'wow' is a palindrome", () => {
>       expect(isPalindrome("wow")).toBe(true);
>     });
> 
>     //Also include failure examples for this high-level behavior
>     it("it knows that 'bill' is NOT a palindrome", () => {
>       expect(isPalindrome("bill")).toBe(false);
>     });
>   });
> });
> `

Wow that's a good catch,
Thanks you!

**Pull request merged**